azure-proto-core/OperationsBase.cs

```diff
@@ -31,9 +32,11 @@ public OperationsBase(ArmClientContext context, Resource resource)
 
         public virtual ResourceIdentifier Id { get; }
 
+        public virtual ArmClientOptions ClientOptions {get; protected set;}
+
         public virtual void Validate(ResourceIdentifier identifier)
         {
-            if (identifier?.Type != ResourceType)
+            if (SubscriptionContainerOperations.AzureResourceType != identifier && identifier?.Type != ResourceType)
```

Comment:
I think this should be

```suggestion
            if (SubscriptionContainerOperations.AzureResourceType != ResourceType && identifier == null)
```

Reply:
I don't think this would work. 

Just tried it real quick with your change suggestion and the subsequent this shouldn't be needed with the other changes

Time Elapsed 00:00:03.36
Unhandled exception. System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter ''Microsoft.Resources/subscriptions' is not a valid resource')
   at azure_proto_core.ResourceIdentifier.Parse(String id) in /mnt/c/Users/stevens/projects/dotnetsdk/azure-proto-sdk/azure-proto-core/Resources/ResourceIdentifier.cs:line 130
   at azure_proto_core.ResourceIdentifier..ctor(String id) in /mnt/c/Users/stevens/projects/dotnetsdk/azure-proto-sdk/azure-proto-core/Resources/ResourceIdentifier.cs:line 32
   at azure_proto_core.ResourceIdentifier.op_Implicit(String other) in /mnt/c/Users/stevens/projects/dotnetsdk/azure-proto-sdk/azure-proto-core/Resources/ResourceIdentifier.cs:line 199
   at azure_proto_core.SubscriptionContainerOperations..ctor(ArmClientContext context, ArmClientOptions options) in /mnt/c/Users/stevens/projects/dotnetsdk/azure-proto-sdk/azure-proto-core/SubscriptionContainerOperations.cs:line 19
   at azure_proto_core.ArmClient.Subscriptions() in /mnt/c/Users/stevens/projects/dotnetsdk/azure-proto-sdk/azure-proto-core/ArmClient.cs:line 62
   at azure_proto_core.ArmClient.GetDefaultSubscription(CancellationToken token) in /mnt/c/Users/stevens/projects/dotnetsdk/azure-proto-sdk/azure-proto-core/ArmClient.cs:line 197
   at azure_proto_core.ArmClient..ctor(Uri baseUri, TokenCredential credential, String defaultSubscriptionId, ArmClientOptions options) in /mnt/c/Users/stevens/projects/dotnetsdk/azure-proto-sdk/azure-proto-core/ArmClient.cs:line 41
   at azure_proto_core.ArmClient..ctor(ArmClientOptions options) in /mnt/c/Users/stevens/projects/dotnetsdk/azure-proto-sdk/azure-proto-core/ArmClient.cs:line 27
   at client.ClientOptionsOverride.Execute() in /mnt/c/Users/stevens/projects/dotnetsdk/azure-proto-sdk/client/Scenarios/ClientOverrides.cs:line 23
   at client.Program.Main(String[] args) in /mnt/c/Users/stevens/projects/dotnetsdk/azure-proto-sdk/client/Program.cs:line 14